src/napari_experimental/layergroup.py

```diff
@@ -0,0 +1,339 @@
+from __future__ import annotations
+
+import itertools
+import warnings
+from typing import TYPE_CHECKING, Iterable, List, Optional
+
+import numpy as np
+
+from napari.components.layerlist import Extent
+from napari_experimental.layerlist import _LayerListMixin
+
+from napari.layers.base import Layer
+from napari.utils.naming import inc_name_count
+from napari.utils.translations import trans
+from napari.utils.tree import Group
+
+if TYPE_CHECKING:
+    from npe2.manifest.contributions import WriterContribution
+
+def combine_extents(extents):
+    """Combine extents into a single extent that bounds all individual ones.
+    Each extent can be cast to an ndim x 2 (min, max) array, but ndim
+    could be different for each extent because of broadcasting. In that case,
+    we need to *prepend* the row (0, 0) to each array so that they match in
+    size.
+    """
+    if len(extents) == 0:
+        return []
+    extent_arrays = [np.array(ex) for ex in extents]
+    ndims = [arr.shape[1] for arr in extent_arrays]
+    required_ndims = max(ndims)
+    dims_to_prepend = [required_ndims - d for d in ndims]
+    padded_arrays = [
+        np.pad(ex, ((0, 0), (d, 0)))
+        for ex, d in zip(extent_arrays, dims_to_prepend)
+    ]
+    big_array = np.stack(padded_arrays, axis=0)
+    result = np.stack(
+        [np.min(big_array[:, 0], axis=0), np.max(big_array[:, 1], axis=0)],
+        axis=1,
+    ).T
+    return result
+
+class LayerGroup(Group[Layer], Layer, _LayerListMixin):
+    def __init__(
+        self, children: Iterable[Layer] = (), name: str = 'LayerGroup'
+    ) -> None:
+        Group.__init__(self, children, name=name, basetype=Layer)
+        Layer.__init__(self, None, self._get_ndim(), name=name)
```

Comment:
```suggestion
        Group.__init__(self, children, name=name, basetype=Layer)
        group_emitters = self.events
        Layer.__init__(self, None, self._get_ndim(), name=name)
        self.events.add(**group_emitters.emitters)
```

This hack on my own branch gets around the fact that `Layer` overwrites the `Group` emitter object, by caching the group one then recombining them afterwards. At least, it does on my `wgraham/playground` branch.